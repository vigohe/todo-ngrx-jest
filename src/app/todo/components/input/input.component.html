<div>
  <input #todo_input placeholder="Add a todo" (keyup.enter)="addTodo(todo_input.value); todo_input.value=''" />
</div>
