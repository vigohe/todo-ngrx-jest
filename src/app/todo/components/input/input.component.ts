import { Component, OnInit, Output, EventEmitter } from '@angular/core';

@Component({
  selector: 'app-input',
  templateUrl: './input.component.html',
  styleUrls: ['./input.component.css']
})
export class InputComponent implements OnInit {

  @Output() add = new EventEmitter<String>();

  constructor() { }

  ngOnInit() {
  }

  addTodo(todoText: String) {
    this.add.emit(todoText);
  }

}
