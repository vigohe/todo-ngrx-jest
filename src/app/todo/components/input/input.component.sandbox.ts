import { sandboxOf } from 'angular-playground';
import { InputComponent } from './input.component';

export default sandboxOf(InputComponent)
  .add('with simple text', {
    template: `<app-input></app-input>`
  });
