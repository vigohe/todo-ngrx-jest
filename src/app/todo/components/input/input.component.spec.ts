import {InputComponent} from './input.component';
import { ComponentFixture, TestBed } from '@angular/core/testing';


describe('InputComponent', () => {
  let component: InputComponent;
  let fixture: ComponentFixture<InputComponent>;

  beforeEach(async() => {
    TestBed.configureTestingModule({
      imports: [],
      declarations: [ InputComponent]
    });

    await TestBed.compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(InputComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });


  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should compile', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
