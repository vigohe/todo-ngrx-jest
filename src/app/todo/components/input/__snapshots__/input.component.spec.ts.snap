// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputComponent should compile 1`] = `
<app-input
  add={[Function EventEmitter]}
>
  <div>
    
    
    <input
      placeholder="Add a todo"
    />
    
  
  </div>
  
</app-input>
`;
